### ISSUE TITLE
Cannot find a connection between any variable and the result of the loss function y=f(x).

### ISSUE THREAD

**Issue by Flossed** [Flossed/llm-mini#25]
training.js:260 Training error: Error: Cannot find a connection between any variable and the result of the loss function y=f(x). Please make sure the operations that use variables are inside the function f passed to minimize().
    at MiniLLMTrainer.trainBatch (training.js:93:30)
    at async MiniLLMTrainer.trainOnText (training.js:40:30)
    at async startTraining (training.js:257:9)

**Comment by Flossed** [comment]
## Fixed! ✅

The error was caused by computing the loss outside of the `minimize()` function, which prevented TensorFlow.js from tracking the gradients properly.

### Problem:
```javascript
// Loss was computed in tf.tidy, then passed to minimize
const loss = await tf.tidy(() => {
    // ... compute loss ...
    return loss;
});
await this.optimizer.minimize(() => loss); // ❌ Loss already computed
```

### Solution:
```javascript
// Now computing loss inside minimize function
const lossValue = await this.optimizer.minimize(() => {
    // Forward pass
    const predictions = this.model.model.predict(inputs);
    
    // Calculate loss
    const loss = tf.losses.softmaxCrossEntropy(targFlat, predFlat);
    
    return loss;
}, true); // ✅ Gradients can be tracked
```

The key changes:
1. Moved all loss computation inside the `minimize()` function
2. Tensor preparation happens outside to avoid recreation
3. Added `true` parameter to return the scalar loss value
4. Proper tensor cleanup after training step

The training should now work! You'll see:
- Loss values decreasing
- Sample text generation every 5 epochs
- Your GPU being utilized for training

Reload and try training again!
